client/src/endToEndManualTest/java/io/obswebsocket/community/client/test/ObsRemoteE2eObservationIT.java

```diff
@@ -1,223 +1,248 @@
-//    obsShould("Fade to scene2");
-//    remote.transitionToProgram(TRANSITION_FADE, 1500, loggingCallback);
-//    obsShould("Disable studio mode");
-//    remote.setStudioModeEnabled(false, loggingCallback);
-//
-//  }
-//
+  }
+
+  @AfterEach
+  public void afterEach() {
+    try {
+      Thread.sleep(1000);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    }
+    System.out.println("<< Test Complete");
+
+  }
+
+  @Test
+  void switchScene() {
+    obsShould("Switch to scene2");
+    remote.setCurrentProgramScene(SCENE2, loggingCallback());
+    obsShould("Switch back to scene1");
+    remote.setCurrentProgramScene(SCENE1, loggingCallback());
+  }
+
+  @Test
+  void showHideSceneItem() {
+    Number squareId = remote.getSceneItemId(SCENE1, SOURCE_RED_SQUARE, 0, 1000)
+        .getSceneItemId();
+
+    obsShould("Show the red square");
+    remote.setSceneItemEnabled(SCENE1, squareId, true, loggingCallback());
+    obsShould("Hide the red square");
+    remote.setSceneItemEnabled(SCENE1, squareId, false, loggingCallback());
+  }
+
+  @Test
+  void setTransition() {
+    obsShould("Set current transition to Slide, with transition 2000ms");
+
+    remote.setCurrentSceneTransition(TRANSITION_SLIDE, loggingCallback());
+    remote.setCurrentSceneTransitionDuration(2000, loggingCallback());
+    obsShould("Set current transition back to 300ms");
+    remote.setCurrentSceneTransitionDuration(300, loggingCallback());
+    obsShould("Set current transition back to Cut");
+    remote.setCurrentSceneTransition(TRANSITION_CUT, loggingCallback());
+  }
+
+  @Test
+  void changeScenesWithTransition() {
+    obsShould("Change to scene2, with the Slide transition");
+    remote.setCurrentSceneTransition(TRANSITION_SLIDE, 1000);
+    remote.setCurrentProgramScene(SCENE2, loggingCallback());
+  }
+
+  @Test
+  void setSourceFilterVisibility() {
+    Number squareId = remote.getSceneItemId(SCENE1, SOURCE_RED_SQUARE, 0, 1000)
+        .getSceneItemId();
+
+    obsShould("Show a blue square (red square colored blue by a filter)");
+    remote.setSceneItemEnabled(SCENE1, squareId, true, loggingCallback());
+    remote.setSourceFilterEnabled(SOURCE_RED_SQUARE, SOURCE_RED_SQUARE_FILTER, true,
+        loggingCallback());
+    obsShould("Return the red square to normal");
+    remote.setSceneItemEnabled(SCENE1, squareId, false, loggingCallback());
+  }
+
+  @Test
+  void exerciseStudioMode() {
+    obsShould("Enable studio mode");
+    remote.setStudioModeEnabled(true, loggingCallback());
+    obsShould("Set preview scene to scene2");
+    remote.setCurrentPreviewScene(SCENE2, loggingCallback());
+    remote.setCurrentSceneTransition(TRANSITION_FADE, loggingCallback());
+
+    obsShould("Fade to scene2");
+    remote.triggerStudioModeTransition(loggingCallback());
+
+    obsShould("Disable studio mode");
+    remote.setStudioModeEnabled(false, loggingCallback());
+  }
+
+//  TODO: I don't see how to change the source name
```

Comment:
SetSceneItemName?

Reply:
The todo might not be clear, it's not the name of the sceneitem that needs changing, it's the name that is displayed. So the text property.

Reply:
SetSceneItemSettings then

Reply:
SetInputSettingsRequest seems to work. Saw that method but thought it wouldn't do what I wanted because these things are called SceneItem and not Sources...